### ISSUE TITLE
Forward selection with method = lmStepAic

### ISSUE THREAD

**Issue by apaust1** [topepo/caret#341]
I've been using stepAIC for a report and was looking to switch over to another method, and this package seemed perfect for my needs, however I am having a hard time implementing it. I am using forward selection, and since my data sets often have hundreds of variables I intend on keeping this method. I noticed that in train's call to AIC there is a default object passed to the function. Is there a way to make it so an object can be passed as 'the initial model in the stepwise search'.

 In line 12 of https://github.com/topepo/caret/blob/master/models/files/lmStepAIC.R  the call to stepAIC is made and the object lm(.outcome~., data=dat) is passed to the argument object. For forward selection can this object be either dynamic or lm(.outcome~1, data=dat). (This is my first post to github if there is any other info that you need, or some other format that you'd prefer let me know please.)

 Thanks, 
Aaron Aust

**Comment by topepo** [comment]
The easiest way to solve this is to used a [custom method](http://topepo.github.io/caret/custom_models.html). You can copy the existing model code using `getModelInfo("lmStepAIC", regex = FALSE)` and then redefine the `fit` code. Instead of:

```r
stepAIC(lm(.outcome ~ ., data = dat), ...)
```

you can use any `lm` model that you like.

**Comment by apaust1** [comment]
Ok thanks. This will fix it for me. While trying some other things I tried passing an initial object like I would if I was using stepAIC directly. 

stepAIC(object, scope, scale = 0,
        direction = c("both", "backward", "forward"),
        trace = 1, keep = NULL, steps = 1000, use.start = FALSE,
        k = 2, ...)

If i pass an argument named object to the train call, instead of it being ignored, or not used,  it gets passed to the next unspecified input i.e. if scale is unspecified it becomes what object was and is no longer the default of 0, which then breaks the next use of scale. I'm not sure how this happens but guess its a bug. This is how I found out about train's call to stepAIC.
